com.unity.shadergraph/Editor/Data/Nodes/Math/Advanced/RefractNode.cs

```diff
@@ -0,0 +1,80 @@
+using System.Reflection;
+using UnityEngine;
+using UnityEditor.Graphing;
+using UnityEditor.ShaderGraph.Drawing.Controls;
+
+namespace UnityEditor.ShaderGraph
+{
+    enum RefractMode
+    {
+        CriticalAngle,
+        Safe
+    };
+
+    [Title("Math", "Advanced", "Refract")]
+    class RefractNode : CodeFunctionNode
+    {
+        [SerializeField]
+        private RefractMode m_RefractMode = RefractMode.Safe;
+
+        [EnumControl("Mode")]
+        public RefractMode refractMode
+        {
+            get { return m_RefractMode; }
+            set
+            {
+                if (m_RefractMode == value)
+                    return;
+
+                m_RefractMode = value;
+                Dirty(ModificationScope.Graph);
+            }
+        }
+
+        public RefractNode()
+        {
+            name = "Refract";
+            synonyms = new string[] { "refract", "refract safe", "refract critical angle" };
```

Comment:
synonyms are alternate search terms, so that people can find the node using alternate names in the search - so you don't need to include the node's name among the synonyms.  I recommend using terms like warp, bend, and distort as synonyms for refract.